core-addon/IonCore.js

```diff
@@ -47,6 +49,25 @@ module.exports = class IonCore {
     };
     browser.management.onInstalled.addListener(addonStateHandler);
     browser.management.onUninstalled.addListener(addonStateHandler);
+
+    // Listen for incoming messages from the studies.
+    //
+    // ***IMPORTANT***:
```

Comment:
Thank you for commenting this thoroughly!

Reply:
Oh, my pleasure and duty! I spent some time figuring it out and asking Addons folks, so it was the least I could do!